test/engine-luatest/gh_4544_collation_drop_test.lua

```diff
@@ -0,0 +1,27 @@
+local t = require('luatest')
+
+local function before_all(cg)
+    local server = require('luatest.server')
+    cg.server = server:new()
+    cg.server:start()
+end
+
+local function after_all(cg)
+    cg.server:drop()
+end
+
+local g1 = t.group('gh-4544-1')
+g1.before_all(before_all)
+g1.after_all(after_all)
+
+-- Check that key_def compare doesn't crash after collation drop.
+g1.test_keydef_compare = function(cg)
+    cg.server:exec(function()
+        local coll_name = 'unicode_af_s1'
+        local key_def = require('key_def')
+        local kd = key_def.new({{field = 1, type = 'string',
+                                 collation = coll_name}})
+        box.internal.collation.drop(coll_name)
```

Comment:
What does `kd:totable()[1].collation` show after this point? What if another collation is created with the same id? Maybe, we could show `coll.fingerprint` if the collation was deleted, including the case when another collation was created instead?

Reply:
> What does kd:totable()[1].collation show after this point?

`./src/box/lua/key_def.c:103: luaT_push_key_def_parts: Assertion `coll_id != NULL' failed.`

Reply:
Fixed.